src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java

```diff
@@ -386,9 +390,38 @@ public LibraryToLink createLibraryLinkerInput(
     Artifact dynamicLibrary = nullIfNone(dynamicLibraryObject, Artifact.class);
     Artifact interfaceLibrary = nullIfNone(interfaceLibraryObject, Artifact.class);
 
-    Artifact notNullArtifactForIdentifier = null;
     StringBuilder extensionErrorsBuilder = new StringBuilder();
     String extensionErrorMessage = "does not have any of the allowed extensions";
+
+    PathFragment dynamicLibraryPathFragment = null;
+    if (dynamicLibraryPath != null && !dynamicLibraryPath.isEmpty()) {
```

Comment:
Use Strings.isNullOrEmpty() instead  (from Guava)
https://guava.dev/releases/23.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-

Reply:
I've replaced these checks by calls to `Strings.isNullOrEmpty()`.